text/0000-mixed-shadow-mode.md

```diff
@@ -0,0 +1,206 @@
+---
+title: Mixed Shadow Mode
+status: DRAFTED
+created_at: 2021-05-07
+updated_at: YYYY-MM-DD
+pr: https://github.com/salesforce/lwc-rfcs/pull/49
+---
+
+# Mixed Shadow Mode
+
+## Introduction
+
+LWC offers a collection of polyfills that implement Shadow DOM features. These polyfills are
+collectively referred to as "synthetic Shadow DOM" and are published to npm under the
+`@lwc/synthetic-shadow` package. Synthetic shadow DOM provides a consistent experience for LWC web
+components across the Salesforce-supported browser matrix.
+
+Native Shadow DOM support has greatly increased since the start of the LWC project but applications
+continue to rely on the synthetic shadow polyfills to maintain backwards-compatibility. The removal
+of these polyfills can result in broken experiences due to dependencies on certain "escape hatches"
+that are not possible in a native Shadow DOM environment (e.g., instrumentation, styling, etc, which
+are discussed below).
+
+Applications that wish to use LWC components in a native shadow context simply need to omit the
+inclusion of the `@lwc/synthetic-shadow` polyfill; however, this is not a realistic option for
+large applications that cannot afford a rewrite.
+
+This proposal opens up an incremental migration path for applications moving to native Shadow DOM by
+allowing the usage of both native and synthetic Shadow DOM in the same document.
+
+## Detailed design
+
+Components will opt-in to this new feature and the default mode will be synthetic mode. From the
+component's perspective, there are two choices of shadow semantics: synthetic mode or native
+mode. The concept of a mixed mode should be transparent and the framework should guarantee this
+through integration testing, which is further discussed below.
+
+A component can signal that it prefers native Shadow DOM by setting a static property called
+`preferNativeShadow` to `true`.
+
+```js
+export default class extends LightningElement {
+    static preferNativeShadow = true;

```

Comment:
what an ugly property name :)

Reply:
maybe a enum: `static defaultShadowMode = 'native'`

Reply:
btw, I realized it is probably too late for me to bikeshed on this... but I can't help myself! lol

Reply:
I like it! I don't know anyone who has a strong preference for `preferNativeShadow`. I'll bring it up with the rest of the team and make the changes if nobody objects.

Reply:
I like it as well.

Reply:
I was just chatting about this with @nolanlawson and we were a little concerned about how similar it sounds to open/closed modes of shadow roots.